### PR TITLE
Pull-request for issue-10

### LINKED ISSUE #10: Add `{% gitcommit short=True %}` option
Currently this library only allows users to get the full commit hash.

I would see some possible usages of the short commit hash as well.
So, I would like to propose the following syntax for it: `{% gitcommit short=True %}`.

What needs to be done?
1. we need to find out how to pass parameters to the `jinja2` tags
2. we need to add a new parameter to the `_commit_hash` method
3. we need to write tests for this feature
4. we need to write docs for this feature as well https://github.com/sobolevn/jinja2-git#usage

This was originally a `# TODO` comment: https://github.com/sobolevn/jinja2-git/blob/23172f3a7f26b6f551f70a8c6aaa818a79918e4f/jinja2_git.py#L22

### PR DESCRIPTION
**Pull request by mikhail-akimov**
Added the ability to receive a short version of the commit.
Closes #10

### CODE REVIEW

**Comment by mikhail-akimov** [comment]
I think it's not compleate. Let me tune it.

File: jinja2_git.py [hunk 1]
```diff
@@ -11,14 +11,27 @@ class GitExtension(Extension):
 
     tags = {'gitcommit'}
 
-    def _commit_hash(self):
-        output = subprocess.check_output(['git', 'rev-parse', 'HEAD'])
+    def _commit_hash(self, *args):
```
Comment (sobolevn) [inline jinja2_git.py]: This should be: `def _commit_hash(self, short=False):`

File: jinja2_git.py [hunk 2]
```diff
@@ -11,14 +11,27 @@ class GitExtension(Extension):
...
+        if args[0]:
+            output = subprocess.check_output(['git',
```
Comment (sobolevn) [inline jinja2_git.py]: Can you please line these items on a single line, like so:
```python
(
    [ 'git', 'rev-parse', '--short', 'HEAD'],
)
```

File: jinja2_git.py [hunk 3]
```diff
@@ -11,14 +11,27 @@ class GitExtension(Extension):
...
+                                              'rev-parse',
+                                              '--short',
+                                              'HEAD',
+                                              ])
+        else:
+            output = subprocess.check_output(['git', 'rev-parse', 'HEAD'])
```
Comment (sobolevn) [inline jinja2_git.py]: Please, try to avoid duplicating the command.

Comment (mikhail-akimov) [reply jinja2_git.py]: What is the best way to avoid duplicating here? Bring the list to a variable?

Comment (sobolevn) [reply jinja2_git.py]: I can thing about `insert()` call when `short is True`.
There are other options as well.

File: jinja2_git.py [hunk 4]
```diff
@@ -11,14 +11,25 @@ class GitExtension(Extension):
...
 
-        # TODO: add {% gitcommit 'short' %} feature
-        result = self.call_method('_commit_hash', [], lineno=lineno)
+        if parser.stream.skip_if('name:short'):
+            parser.stream.skip(1)
+            short = parser.parse_expression()
+        else:
+            short = nodes.Const(False)
+
+        args = [short]
```
Comment (sobolevn) [inline jinja2_git.py]: See my previous note about `args`.

File: jinja2_git.py [hunk 5]
```diff
@@ -11,14 +11,25 @@ class GitExtension(Extension):
...
+        result = self.call_method('_commit_hash', args, [], lineno=lineno)
```
Comment (sobolevn) [inline jinja2_git.py]: It should be: `self.call_method('_commit_hash', [short], [], lineno=lineno)`

**Comment by mikhail-akimov** [comment]
I getting short=False twice (1-st in parse() and second at _commit_hash() as default value). Can i optimize it somehow?

**Comment by sobolevn** [comment]
@mikhail-akimov remove the default value from `_commit_hash` function.

**Comment by sobolevn** [comment]
@mikhail-akimov awesome work, thanks!

**Pull request merged**